tensorflow-framework/src/main/java/org/tensorflow/framework/activations/ReLU.java

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * Rectified Linear Unit(ReLU) activation.
+ *
+ * <p>With default values, this returns the standard ReLU activation: <code>max(x, 0)</code>, the
+ * element-wise maximum of 0 and the input tensor.
+ *
+ * <p>Modifying default parameters allows you to use non-zero thresholds, change the max value of
+ * the activation, and to use a non-zero multiple of the input for values below the threshold.
+ *
+ * <p>For example:
+ *
+ * <pre>
+ *     Operand&lt;TFloat32&gt; input = tf.constant(
+ *              new float[] {-10f, -5f, 0.0f, 5f, 10f});
+ *
+ *     // With default parameters
+ *     ReLU&lt;TFloat32&gt; relu = new ReLU&lt;&gt;(tf);
+ *     Operand&lt;TFloat32&gt; result = relu.call(input);
+ *     // result is [0.f,  0.f,  0.f,  5.f, 10.f]
+ *
+ *     // With alpha = 0.5
+ *     relu = new ReLU&lt;&gt;(tf, 0.5f, ReLU.MAX_VALUE_DEFAULT, ReLU.THRESHOLD_DEFAULT);
+ *     result = relu.call(input);
+ *     // result is [-5.f , -2.5f,  0.f ,  5.f , 10.f]
+ *
+ *     // With maxValue = 5
+ *     relu = new ReLU&lt;&gt;(tf, ReLU.ALPHA_DEFAULT, 5f, ReLU.THRESHOLD_DEFAULT);
+ *     result = relu.call(input);
+ *     // result is [0.f, 0.f, 0.f, 5.f, 5.f]
+ *
+ *     // With threshold = 5
+ *     relu = new ReLU&lt;&gt;(tf, ReLU.ALPHA_DEFAULT, ReLU.MAX_VALUE_DEFAULT, 5f);
+ *     result = relu.call(input);
+ *     // result is [-0.f, -0.f,  0.f,  0.f, 10.f]
+ * </pre>
+ *
+ * @param <T> the data type of the result
+ */
+public class ReLU<T extends TNumber> extends Activation<T> {
+
+  public static final float ALPHA_DEFAULT = 0.0f;
+  public static final float MAX_VALUE_DEFAULT = Float.NaN;
+  public static final float THRESHOLD_DEFAULT = 0.0f;
+
+  private final float alpha;
+  private final float maxValue;
+  private final float threshold;
+
+  /**
+   * Creates a new ReLU with alpha={@link #ALPHA_DEFAULT}, maxValue={@link #MAX_VALUE_DEFAULT},
+   * threshold={@link #THRESHOLD_DEFAULT},
+   *
+   * @param tf the TensorFlow Ops
+   */
+  public ReLU(Ops tf) {
+    this(tf, ALPHA_DEFAULT, MAX_VALUE_DEFAULT, THRESHOLD_DEFAULT);
+  }
+
+  /**
+   * Creates a new ReLU
+   *
+   * @param tf the TensorFlow Ops
+   * @param alpha governs the slope for values lower than the threshold.
+   * @param maxValue sets the saturation threshold (the largest value the function will return).
+   * @param threshold the threshold value of the activation function below which values will be
+   *     damped or set to zero.
+   */
+  public ReLU(Ops tf, float alpha, float maxValue, float threshold) {
+    super(tf);
+    this.alpha = alpha;
+    this.maxValue = maxValue;
+    this.threshold = threshold;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Operand<T> call(Operand<T> input) {
+
+    DataType<T> dataType = input.asOutput().dataType();
+
+    boolean clipMax = !Float.isNaN(maxValue);
+    Operand<T> negativePart = null;
+    if (alpha != 0) {
+      if (Float.isNaN(maxValue) && threshold == 0) {
+        // TODO LeakyRelu is not in tf.nn ????
```

Comment:
`LeakyRelu` is not add to `tf.nn` because it is set as [not visible](https://github.com/tensorflow/tensorflow/blob/465eab955687d78dc7431fcc52e62d7694d2c6da/tensorflow/core/api_def/base_api/api_def_LeakyRelu.pbtxt#L3) in the main API def proto. To add it to `tf.nn`, you can override its visibility by setting it to `VISIBLE` in [our proto](https://github.com/tensorflow/java/blob/master/tensorflow-core/tensorflow-core-api/src/bazel/api_def/api_def_LeakyRelu.pbtxt)

Reply:
@karllessard should we do this or just leave it as is for internal use? In Python it is available as `tf.nn.leaky_relu`.
If we want to add this, it may be better as a separate PR because of the generated classes being changed.

Reply:
I suggest we make it visible in this PR, even if the ops will be regenerated, only two files (ˋNnOps` and ˋLeakyRelu`) should end up being different

Reply:
I have made the changes to add `LeakyRelu` to NnOps and generate `tf.nn.leakyRelu()`